### ISSUE TITLE
Transactional API at client side

### ISSUE THREAD

**Issue by sbernard31** [eclipse/leshan#968]
Bootstap operation and write composite need atomic modification of the LWM2M tree.
So maybe we need a commit/rollback mechanism or at least an API which allow user to implement it.

**Comment by sbernard31** [comment]
@moznion, do you already feel you need something like this ?

**Comment by moznion** [comment]
Yes, absolutely! Sometimes I was feeling I wish I were able to manipulate the tree atomically.

**Comment by sbernard31** [comment]
Do you already have a vague idea about what the API should looks like ?

**Comment by sbernard31** [comment]
This could maybe also used to support a more secure bootstrap session. (see https://github.com/OpenMobileAlliance/OMA_LwM2M_for_Developers/issues/523)
